### PR TITLE
feat(core): detect and report file system type for the database location

### PR DESCRIPTION
**Pull request by bluestreak01**
We would like to avoid situation where users unknowingly run QuestDB on file systems that we do not support. Such as NFS, CIFS, SMB etc. QuestDB will now alert user in the log that filesystem is not supported, but will not abort the startup sequence.

Since I could not build on FreeBSD I replaced packaging mechanism. It used to be NodeJS, but since this was the part allegedly failing it is now done via `pom.xml` and assembly plugin.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 10 / 30 (33.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|9|00.00%|
|:large_blue_circle:|io/questdb/std/str/Path.java|0|7|00.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|11|90.91%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 10 / 31 (32.26%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|10|00.00%|
|:large_blue_circle:|io/questdb/std/str/Path.java|0|7|00.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|11|90.91%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 10 / 31 (32.26%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|10|00.00%|
|:large_blue_circle:|io/questdb/std/str/Path.java|0|7|00.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|11|90.91%|

File: core/src/main/java/io/questdb/ServerMain.java [hunk 1]
```diff
@@ -125,6 +125,20 @@ public ServerMain(String[] args) throws Exception {
                 break;
         }
         log.advisory().$("available CPUs: ").$(Runtime.getRuntime().availableProcessors()).$();
+        log.advisory().$("db root: ").$(configuration.getCairoConfiguration().getRoot()).$();
```
Comment (ideoma) [inline core/src/main/java/io/questdb/ServerMain.java:128]: shall we check backup path too?

File: core/src/main/java/io/questdb/std/Files.java [hunk 2]
```diff
@@ -138,10 +146,36 @@ public static long getOpenFileCount() {
...
+     * Detects if filesystem is supported by QuestDB. The function returns both FS magic and name. Both
+     * can be presented to user even if file system is not supported.
+     *
+     * @param lpszName existing path on the file system. The name of the filesystem is written to this
+     *                 address, therefore name should have at least 128 byte capacity
+     * @return 0 when OS call failed, errno should be checked. Negative number is file system magic that is supported
+     * positive number is magic that is not supported.
+     */
+    public static int getFileSystemStatus(LPSZ lpszName) {
+        return getFileSystemStatus(lpszName.address());
```
Comment (ideoma) [inline core/src/main/java/io/questdb/std/Files.java:176]: I'd ensure lpszName has enough capacity to accept return string with an `assert` or `if`

File: core/src/main/java/io/questdb/ServerMain.java [hunk 3]
```diff
@@ -125,6 +125,20 @@ public ServerMain(String[] args) throws Exception {
...
+        try (Path path = new Path()) {
+            path.of(configuration.getCairoConfiguration().getRoot()).$();
```
Comment (ideoma) [inline core/src/main/java/io/questdb/ServerMain.java:130]: There is dependency here that path will be 255 chars and implementation of that can change in the future.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 32 (62.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|10|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|11|90.91%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 32 (62.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|10|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|11|90.91%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 32 (62.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|10|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|11|90.91%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 32 (62.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|10|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|11|90.91%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

File: core/src/main/java/io/questdb/ServerMain.java (hunk 3 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/ServerMain.java:130]: Path is currently allocated 256 bytes unless capacity is specified lower:

image

There is no dependency between path to the database and returned file system name

File: core/src/main/java/io/questdb/ServerMain.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/ServerMain.java:128]: done here https://github.com/questdb/questdb/pull/1725/commits/444239783f1a60cd1886777a8eb03f12e9b458cf

File: core/src/main/java/io/questdb/std/Files.java (hunk 2 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/std/Files.java:177]: done here: https://github.com/questdb/questdb/pull/1725/commits/444239783f1a60cd1886777a8eb03f12e9b458cf

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 40 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|15|00.00%|
|:large_blue_circle:|io/questdb/std/str/CharSequenceZ.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|12|83.33%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 40 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|15|00.00%|
|:large_blue_circle:|io/questdb/std/str/CharSequenceZ.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|12|83.33%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 40 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|15|00.00%|
|:large_blue_circle:|io/questdb/std/str/CharSequenceZ.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|12|83.33%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 40 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|15|00.00%|
|:large_blue_circle:|io/questdb/std/str/CharSequenceZ.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|12|83.33%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 40 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|15|00.00%|
|:large_blue_circle:|io/questdb/std/str/CharSequenceZ.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|12|83.33%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 40 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|15|00.00%|
|:large_blue_circle:|io/questdb/std/str/CharSequenceZ.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|12|83.33%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 40 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|15|00.00%|
|:large_blue_circle:|io/questdb/std/str/CharSequenceZ.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|12|83.33%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 20 / 40 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|15|00.00%|
|:large_blue_circle:|io/questdb/std/str/CharSequenceZ.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/log/NullLogRecord.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/std/Files.java|10|12|83.33%|
|:large_blue_circle:|io/questdb/std/str/Path.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/log/Logger.java|2|2|100.00%|

**Review by ideoma — APPROVED** [review]

**Pull request merged**